tests/Middleware/RateLimiterTest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+use Envato\Middleware\RateLimiter;
+use PHPUnit\Framework\TestCase;
+
+final class RateLimiterTest extends TestCase {
+  private $rate_limiter;
+
+  public function setUp() {
+    $this->rate_limiter = new RateLimiter();
+  }
+
+  public function tearDown() {
+    // Don't allow this lock file to be carried between tests.
+    if (file_exists($this->rate_limiter->full_system_path_to_lock_file())) {
+      unlink($this->rate_limiter->full_system_path_to_lock_file());
+    }
+  }
+
+  public function testIsThrottled() {
+    $throttled = $this->rate_limiter->setThrottle(1);
+    $this->assertTrue($this->rate_limiter->isThrottled());
+  }
+
+  public function testThrottlerIsRemovedAfterTimeout() {
+    $throttled = $this->rate_limiter->setThrottle(0);
+    $this->assertFileExists($this->rate_limiter->full_system_path_to_lock_file());
+    sleep(1); // This is bad and needs some time travelling lib
```

Comment:
Like the comment says, we need to move to something like Ruby's [`travisjeffery/timecop`](https://github.com/travisjeffery/timecop) to avoid spending real time waiting for time to elapse. I don't think it's a major issue right now but will be something that needs addressing as this suite grows.